Automattic/wp-calypso
# Add useSubscriptionManagerSiteDeliveryFrequencyMutation

Closes [#74965](https://github.com/Automattic/wp-calypso/issues/74965)

## Proposed Changes

This PR adds the `useSubscriptionManagerSiteDeliveryFrequencyMutation()` hook. It changes the delivery frequency for a site subscription.

## Testing Instructions



⚠️ This PR needs a subkey cookie set. You can find my subkey in the patch above. Navigate to `http://calypso.localhost:3000` and set it using the Application tab from Chrome Devtools.  ⚠️

To be able to test this hook you need to add it somewhere to your code. I suggest adding it to `client/landing/subscriptions/user-settings/user-settings.tsx` and calling the hook from there:

```js
	const { mutate: mutateDeliveryFrequency } =
		Reader.useSubscriptionManagerSiteDeliveryFrequencyMutation();
	useEffect( () => {
		mutateDeliveryFrequency( { blog_id: '105013038', delivery_frequency: 'daily' } );
	}, [ mutateDeliveryFrequency ] );

```

Replace 105013038 to the `blogId` of a blog your external user is subscribed to. 

In Devtools > Network you should see a call to the endpoint

<img width="1011" alt="CleanShot 2023-03-29 at 15 03 43@2x" src="https://user-images.githubusercontent.com/528287/228544280-d8a68c7a-518c-44bc-a284-ceae229eb041.png">

If you were subscribed to the blog, you should see success in the response:

<img width="967" alt="CleanShot 2023-03-29 at 15 05 05@2x" src="https://user-images.githubusercontent.com/528287/228544476-b8c782a3-1ee5-410b-926b-cb6b3923d023.png">

This PR also contains a simple test, to check if the mutation calls the API with the expected parameters. This test file will get more integration tests, to see if this mutation invalidates the right query, in a future PR

## Pre-merge Checklist
